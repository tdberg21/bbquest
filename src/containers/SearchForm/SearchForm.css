.search-form-container {
  background-image: url(https://www.pigrigbbq.com/files/2017-02/background.jpg);
  height: 100vh;
  padding-top: 1rem;
  width: 101%;
  background-size: cover;
}

.search-form {
  margin: 2rem auto;
  width: 50%;
  display: flex;
  justify-content: center;
  position: relative;
  top: -94px;
}

.location-field {
  width: 60%;
  height: 1rem;
  color: #333;
  font-size: 1em;
  font-weight: 500;
  padding: .5rem;
  border: none;
}

.submit-location-button {
  height: 2rem;
  background-color: red;
  color: #fff;
  border: none;
  padding: 0.5rem;
  font-size: 1em;

}