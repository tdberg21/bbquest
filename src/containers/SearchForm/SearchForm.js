import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchRestaurantData } from '../../helpers/apiCalls';
import { apiKey } from '../../helpers/apiKey';
import { scrubRestaurants } from '../../helpers/dataCleaners';
import { addRestaurants } from '../../actions';


class SearchForm extends Component {
  constructor() {
    super();

    this.state = {
      location: ''
    };
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const restaurants = await fetchRestaurantData(apiKey, this.state.location);
    const cleanRestaurants = scrubRestaurants(restaurants);
    this.props.addRestaurants(cleanRestaurants);
    this.props.history.push('/restaurants');
  }


  render() {
    return(
      <div>
        <form
          className='login-form'
          onSubmit={this.handleSubmit}
        >
          <input
            className='location-field'
            aria-label='Enter Your Location'
            placeholder='Enter a city and state to find BBQ!'
            type='text'
            name='location'
            value={this.state.location}
            onChange={this.handleChange}
          />
          <button
            className='submit-location-button'
            aria-label='Submit location'
          >
            Submit
          </button>
        </form>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  restaurants: state.restaurants
});

export const mapDispatchToProps = (dispatch) => ({
  addRestaurants: (restaurants) => dispatch(addRestaurants(restaurants))
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchForm);