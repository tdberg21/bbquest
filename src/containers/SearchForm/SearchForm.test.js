import React from 'react';
import { shallow } from 'enzyme';
import { SearchForm } from './SearchForm';

describe('SEARCHFORM TESTS', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<SearchForm />);

    expect(wrapper).toMatchSnapshot();
  });
});