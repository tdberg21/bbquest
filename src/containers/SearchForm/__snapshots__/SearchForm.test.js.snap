// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEARCHFORM TESTS should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="login-form"
        onSubmit={[Function]}
>
        <input
                aria-label="Enter Your Location"
                className="location-field"
                name="location"
                onChange={[Function]}
                placeholder="Enter a city and state to find BBQ!"
                type="text"
                value=""
        />
        <button
                aria-label="Submit location"
                className="submit-location-button"
        >
                Submit
        </button>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            aria-label="Enter Your Location"
            className="location-field"
            name="location"
            onChange={[Function]}
            placeholder="Enter a city and state to find BBQ!"
            type="text"
            value=""
/>,
          <button
            aria-label="Submit location"
            className="submit-location-button"
>
            Submit
</button>,
        ],
        "className": "login-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Enter Your Location",
            "className": "location-field",
            "name": "location",
            "onChange": [Function],
            "placeholder": "Enter a city and state to find BBQ!",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Submit location",
            "children": "Submit",
            "className": "submit-location-button",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="login-form"
          onSubmit={[Function]}
>
          <input
                    aria-label="Enter Your Location"
                    className="location-field"
                    name="location"
                    onChange={[Function]}
                    placeholder="Enter a city and state to find BBQ!"
                    type="text"
                    value=""
          />
          <button
                    aria-label="Submit location"
                    className="submit-location-button"
          >
                    Submit
          </button>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              aria-label="Enter Your Location"
              className="location-field"
              name="location"
              onChange={[Function]}
              placeholder="Enter a city and state to find BBQ!"
              type="text"
              value=""
/>,
            <button
              aria-label="Submit location"
              className="submit-location-button"
>
              Submit
</button>,
          ],
          "className": "login-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "Enter Your Location",
              "className": "location-field",
              "name": "location",
              "onChange": [Function],
              "placeholder": "Enter a city and state to find BBQ!",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "Submit location",
              "children": "Submit",
              "className": "submit-location-button",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
